src/lib/components/Swap/Input.tsx

```diff
@@ -27,31 +25,41 @@ interface InputProps {
 }
 
 export default function Input({ disabled }: InputProps) {
-  const input = useAtomValue(inputAtom)
-  const setValue = useUpdateInputValue(inputAtom)
-  const setToken = useUpdateInputToken(inputAtom)
-  const balance = mockCurrencyAmount
+  const {
+    currencyBalances: { [Field.INPUT]: balance },
+    currencyAmounts: { [Field.INPUT]: inputCurrencyAmount },
+  } = useSwapInfo()
+  const inputUSDC = useUSDCValue(inputCurrencyAmount)
+
+  const [typedInputAmount, updateTypedInputAmount] = useInputAmount()
+  const [inputCurrency, updateInputCurrency] = useInputCurrency()
+
+  //TODO(ianlapham): mimic logic from app swap page
+  const mockApproved = true
 
   return (
-    <InputColumn gap={0.5} approved={input.approved !== false}>
+    <InputColumn gap={0.5} approved={mockApproved}>
       <Row>
         <ThemedText.Subhead2 color="secondary">
           <Trans>Trading</Trans>
         </ThemedText.Subhead2>
       </Row>
       <TokenInput
-        input={input}
+        currency={inputCurrency}
+        amount={(typedInputAmount !== undefined ? typedInputAmount : inputCurrencyAmount?.toSignificant(6)) ?? ''}
         disabled={disabled}
-        onMax={balance ? () => setValue(1234) : undefined}
-        onChangeInput={setValue}
-        onChangeToken={setToken}
+        onMax={balance ? () => updateTypedInputAmount(balance.toExact()) : undefined}
+        onChangeInput={(val) => updateTypedInputAmount(val ?? '')}
+        onChangeCurrency={(currency: Currency) => updateInputCurrency(currency)}
       >
         <ThemedText.Body2 color="secondary">
           <Row>
-            {input.usdc ? `~ $${input.usdc.toLocaleString('en')}` : '-'}
+            {inputUSDC ? `~ $${inputUSDC.toFixed(2)}` : '-'}
             {balance && (
-              <ThemedText.Body2 color={input.value && balance.lessThan(input.value) ? 'error' : undefined}>
-                Balance: <span style={{ userSelect: 'text' }}>{balance.toExact()}</span>
+              <ThemedText.Body2
+                color={inputCurrencyAmount && inputCurrencyAmount.greaterThan(balance) ? 'error' : undefined}
```

Comment:
`inputCurrencyAmount?.greaterThan(balance) ? 'error' : undefined`

Reply:
i think here we only want to show the error if the InputCurrency amount is defined, aka we dont want to show the error state if they havent typed anythign yet

Reply:
:+1:, but `Boolean(undefined) === false`, so if `inputCurrencyAmount` is undefined, this should not show an error state.